### PR TITLE
[DOCS] Applications REST API Update

### LINKED ISSUE #5628: [REST API] Update all Application's Endpoints
#### Current Behavior
A REST API docs update is needed to clarify undeploy behavior.

- Update application endpoints wrt to types and download.

---
#### Contributor [Guides](https://docs.meshery.io/project/contributing) and Resources
- 🛠 [Meshery Build & Release Strategy](https://docs.meshery.io/project/build-and-release)
- 📚 [Instructions for contributing to documentation](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#documentation-contribution-flow)
   - Meshery documentation [site](https://docs.meshery.io/) and [source](https://github.com/meshery/meshery/tree/master/docs)
- 🎨 Wireframes and designs for Meshery UI in [Figma](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI)
- 🙋🏾🙋🏼 Questions: [Layer5 Discussion Forum](https://discuss.layer5.io) and [Layer5 Community Slack](http://slack.layer5.io)

### PR DESCRIPTION
**Pull request by Chadha93**
Signed-off-by: Gaurav Chadha <gauravchadha1676@gmail.com>

**Description**

- [x] Adds the new API docs for types - `/api/application/types `
- [x] Adds the new API docs for download -` /api/application/download/{id}`
- [x] Removed the API docs deploy/undeploy for applications because currently patterns deploy is being used.

This PR fixes #5628 

**Notes for Reviewers**

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by l5io** [comment]
🚀 Preview for commit 024d0846556e369df8512ddde898792ad6eea53b at: https://62e10cd34b5a3972c593fdc7--meshery-docs-preview.netlify.app

**Comment by Chadha93** [comment]
Also update this https://docs.google.com/spreadsheets/d/1ABJCbfQRi0uN_YoP2kmHZ-lTI4S4QuvRk1_unomKNRE/edit#gid=1241442399

File: docs/_data/swagger.yml [hunk 1]
```diff
@@ -296,6 +227,65 @@ paths:
...
+                type: helm chart
+                format: .tgz
+              Docker Compose:
+                type: docker compose
+                format: ".yaml .yml"
+              Kubernetes Manifest:
+                type: kubernetes manifest
+                format: ".yaml .yml"
+  
+  /api/application/download/{id}:
```
Comment (MUzairS15) [inline docs/_data/swagger.yml:260]: /api/application/download/{id}?source-type={source_type}

**Review by MUzairS15 — COMMENTED** [review]
@Chadha93 `/api/application?source-type={source-type}` for saving an application (POST), this also needs to be updated.

**Comment by warunicorn19** [comment]
Would this update also be needed in the in the swagger docs?

**Comment by Chadha93** [comment]
> Would this update also be needed in the in the swagger docs?

I guess not..., should it be? I'll take a look again.

**Comment by leecalcote** [comment]
> Also update this https://docs.google.com/spreadsheets/d/1ABJCbfQRi0uN_YoP2kmHZ-lTI4S4QuvRk1_unomKNRE/edit#gid=1241442399

@Revolyssup is this up to date? No graphql support?

**Comment by Chadha93** [comment]
> 

@Revolyssup I've updated the REST API docs you can review, could you also provide the description update need to be updated in this pr?

**Comment by l5io** [comment]
🚀 Preview for commit e394bba107935dbaef9e088dae4cc2af136c9044 at: https://62eb8ac2b63c2d71d71da0f1--meshery-docs-preview.netlify.app

File: docs/_data/swagger.yml [hunk 2]
```diff
@@ -400,6 +400,65 @@ paths:
...
+      - type: source-type
+        description: query parameter
+        name: application_file
+        in: path
+        required: true
+      responses:
+        "200":
+          description: Fetches a single Meshery Application
+          schema:
+            description: "Returns all Meshery Application based on source type"
```
Comment (leecalcote) [inline docs/_data/swagger.yml:420]:
```suggestion
            description: "Returns all Meshery Application based on source type (original type of application packaging format)"
```

**Review by leecalcote — APPROVED** [review]

**Pull request merged**